specs/unified_kernel_image.md

```diff
@@ -159,3 +159,22 @@ that tools staging or consuming UKIs have a common place to store and look for t
 
 The installed UKIs should have a filename `<version format specification>.efi`, i.e. the filename is left to
 implementers but must be valid for comparisons according to the [Version Format Specification](version_format_specification.md).
+
+UKI addons (cmdline PE addons, systemd-sysext extensions for the initrd) built centrally by distributions and
+installed via package manager should be installed into locations depending on whether they are global UKI addons,
+that should be applied to all UKIs installed in the ESP, or whether they should only be applied to a
+single specific UKI.
+
+Global UKI addons should be installed into `/usr/lib/modules/extra.d/`. UKI-specific addons should be installed
+to `/usr/lib/modules/$UNAME/$UKI.efi.extra.d/`, where `$UNAME` is the output of `uname -r` of the kernel included
+in the UKI and `$UKI` is the name of the corresponding centrally built UKI with the `.efi` extension stripped.
+
+The installed UKI addons should have a filename `<version format specification>.addon.efi`, i.e. the filename
+is left to implementers but must be valid for comparisons according to the
+[Version Format Specification](version_format_specification.md).
+
+### Example
+Given `UNAME="6.9.1-1-foo"` and `UKI="bar.efi"`:
+* UKI will be in `/usr/lib/modules/6.9.1-1-foo/bar.efi`
+* Global addons will be in `/usr/lib/modules/extra.d/`
+* Local addons will be in `/usr/lib/modules/6.9.1-1-foo/bar.efi.extra.d/`
```

Comment:
Sorry if I am asking an already answered question!  But is there a place for user defined global and local addons?

For example, if an user want to add a new cmdline PE addons, that for example enable some debug output? I am also thinking on immutable systems, where the user cannot (should not) change the rootfs content (outside the package manager).

Reply:
The idea is to have all rootfs addons in these two folders as agreed location. You can put your addons anywhere you want in the root fs, they will be considered only if they are present in the ESP.
So if  the user wants to create its own debug output, and the system is immutable, the addon can stay anywhere he wants, as long as it is then copied in the ESP when needed.